packages/module-services/src/module.ts

```diff
@@ -0,0 +1,58 @@
+import type { Module } from '@equinor/fusion-framework-module';
+import type { HttpModule } from '@equinor/fusion-framework-module-http';
+import type {
+    ServiceDiscoveryModule,
+    ServiceDiscoveryProvider,
+} from '@equinor/fusion-framework-module-service-discovery';
+
+import { IApiConfigurator, ApiConfigurator } from './configurator';
+import { IApiProvider, ApiProvider } from './provider';
+
+export type ServicesModuleKey = 'services';
+
+export type ServicesModule = Module<
+    ServicesModuleKey,
+    IApiProvider,
+    IApiConfigurator,
+    [HttpModule, ServiceDiscoveryModule]
+>;
+
+/**
+ *  Configure http-client
+ */
+export const module: ServicesModule = {
+    name: 'services',
+    configure: () => new ApiConfigurator(),
+    initialize: async ({ ref, config, requireInstance, hasModule }) => {
+        if (!config.createClient) {
+            const http = await requireInstance('http');
+            const serviceDiscovery: ServiceDiscoveryProvider | undefined = hasModule(
+                'serviceDiscovery'
+            )
+                ? await requireInstance('serviceDiscovery')
+                : ref?.serviceDiscovery;
+
+            config.createClient = async (name: string) => {
+                if (http.hasClient(name)) {
```

Comment:
Må si jeg ikke er noe fan av nestet if. hva med en helper function her `createClient(name, http, serviceDiscovery)` lite å si for koden men gjør det mer lesbart.

Reply:
this is not nested, there a if and a else if (since the first has a return statement), same as a switch(true)